File: Pandacap.Data/ActivityPubInboundActivity.cs
using System.ComponentModel.DataAnnotations;

namespace Pandacap.Data
{
    /// <summary>
    /// Another ActivityPub actor's interaction with a Pandacap post.
    /// </summary>
    public class ActivityPubInboundActivity
    {
        /// <summary>
        /// An internal ID.
        /// </summary>
        [Key]
        public Guid Id { get; set; }

        /// <summary>
        /// The ID of the activity.
        /// </summary>
        [Required]
        public string ActivityId { get; set; } = "";

        /// <summary>
        /// The type of the activity (e.g. Like, Announce).
        /// </summary>
        [Required]
        public string ActivityType { get; set; } = "";

        /// <summary>
        /// The ID of the post that was interacted with.
        /// </summary>
        public Guid DeviationId { get; set; }

        /// <summary>
        /// The date/time at which this interaction was added.
        /// </summary>
        public DateTimeOffset AddedAt { get; set; }

        /// <summary>
        /// The ID of the actor who interacted with the post.
        /// </summary>
        [Required]
        public string ActorId { get; set; } = "";
    }
}

File: Pandacap.Data/ActivityPubInboundActivity.fs
namespace Pandacap.Data

open System

/// Another ActivityPub actor's interaction with a Pandacap post.
type ActivityPubInboundActivity() =
    member val Id = Guid.Empty with get, set
    member val ActivityId = "" with get, set
    member val ActivityType = "" with get, set
    member val DeviationId = Guid.Empty with get, set
    member val AddedAt = DateTimeOffset.MinValue with get, set
    member val AcknowledgedAt = nullDateTimeOffset with get, set
    member val ActorId = "" with get, set
    member val Username = nullString with get, set
    member val Usericon = nullString with get, set
